src/ad-services/ias-publisher-optimization/index.ts

```diff
@@ -0,0 +1,101 @@
+import { context, utils } from '@ad-engine/core';
+
+const logGroup = 'ias-publisher-optimization';
+const scriptUrl = '//cdn.adsafeprotected.com/iasPET.1.js';
+const SLOT_LIST = [
+	'hivi_leaderboard',
+	'top_leaderboard',
+	'top_boxad',
+	'incontent_boxad_1',
+	'bottom_leaderboard',
+];
+
+interface IasPetConfig {
+	queue?: IasPetQueueElement[];
+	pubId?: string;
+}
+
+interface IasPetQueueElement {
+	adSlots: number[][];
+	dataHandler: (string) => void;
+}
+
+interface IasTargetingData {
+	fr?: string;
+	slots?: any[];
+}
+
+class IasPublisherOptimization {
+	private isLoaded = false;
+	private iasPET: IasPetConfig;
+
+	private isEnabled(): boolean {
+		return context.get('services.iasPublisherOptimization.enabled');
+	}
+
+	load(): void {
+		if (!this.isEnabled()) {
+			utils.logger(logGroup, 'disabled');
+			return;
+		}
+
+		if (!this.isLoaded) {
+			utils.logger(logGroup, 'loading');
+			utils.scriptLoader
+				.loadScript(scriptUrl, 'text/javascript', true, 'first')
+				.then(() => this.setup());
+			this.isLoaded = true;
+		}
+	}
+
+	private setup(): void {
+		// @ts-ignore
+		this.iasPET = __iasPET || {};
+		this.iasPET.queue = this.iasPET.queue || [];
+		this.iasPET.pubId = '930616';
+		this.setInitialTargeting();
+
+		window.googletag.cmd.push(() => {
+			const slots = [];
+			const iasPETSlots = [];
+			SLOT_LIST.forEach((slotName) => {
+				slots.push(context.get(`slots.${slotName}`));
+			});
+
+			slots.forEach((slot) => {
+				const sizes = slot.sizes && slot.sizes.length ? slot.sizes[0].sizes : slot.defaultSizes;
+				const adUnitPath = utils.stringBuilder.build(slot.adUnit || context.get('adUnitId'), {
+					slotConfig: slot,
+				});
+				iasPETSlots.push({
+					adSlotId: slot['adProduct'],
+					size: sizes,
+					adUnitPath,
+				});
+			});
+
+			this.iasPET.queue.push({
+				adSlots: iasPETSlots,
+				dataHandler: this.iasDataHandler,
```

Comment:
```suggestion
				dataHandler: () => this.iasDataHandler(),
```
please, so we're able to use `this` in the function in future